src/zfstools/connection.py

```diff
@@ -70,8 +73,13 @@ def destroy_recursively(self, name):
         subprocess.check_call(self.command + ["destroy", '-r', name])
         self._dirty = True
 
-    def snapshot_recursively(self,name,snapshotname):
-        subprocess.check_call(self.command + ["snapshot", "-r", "%s@%s" % (name, snapshotname)])
+    def snapshot_recursively(self,name,snapshotname,properties={}):
+        plist = sum( map( lambda x: ['-o', '%s=%s' % x ], properties.items() ), [] )
+        subprocess.check_call(self.command + ["snapshot", "-r" ] + plist + [ "%s@%s" % (name, snapshotname)])
+        self._dirty = True
+
+    def set_properties(self,properties):
```

Comment:
Yes, it was added to keep the public API consistent, but there are no consumers. The idea was that there could be more than one consumer of ZFSConnection that might want their own properties. I think an 'add_properties' method that removes duplicates might be more appropriate. In any case, it's currently academic as the functionality is unused, so I'll drop it for now.